aiida_common_workflows/workflows/relax/siesta/generator.py

```diff
@@ -246,10 +261,9 @@ def _get_kpoints(self, key, structure):
                 kpoints_mesh.set_kpoints_mesh_from_density(distance=kp_dict['distance'], offset=kp_dict['offset'])
             else:
                 kpoints_mesh.set_kpoints_mesh_from_density(distance=kp_dict['distance'])
-        else:
-            kpoints_mesh = None
+            return kpoints_mesh
 
-        return kpoints_mesh
+        return None
```

Comment:
Is this dead code or is it actually a valid code path where this return `None`?

Reply:
It's valid. It means that no kpoints specifications are in the protocol and therefore no Kpoints will be set for the Siesta relaxation. This is perfectly fine (for instance molecules).